commons-numbers-complex-arrays/src/main/java/org/apache/commons/numbers/complex/arrays/ComplexList.java

```diff
@@ -311,7 +324,7 @@ private static class ComplexInterleavedList extends ComplexList {
          * @throws IllegalArgumentException if the specified double array doesn't have an even amount of doubles.
          */
         ComplexInterleavedList(double[] fromArray) {
-            if (fromArray.length % 2 != 0) {
+            if ((fromArray.length & 1) != 0) {
                 throw new IllegalArgumentException("Length of array has to be even");
             }
             realAndImagParts = fromArray;
```

Comment:
Here the data is not defensively cloned. This is fine but the behaviour should be documented. Any public method should indicate that the argument array is used in-place; any external modifications to the array will be reflected in this List until a structural modification is made to the data storage, for example a resize.